File: frontend/src/services/inspeccionService.ts
import api from './api'
import type { IInspeccion, IInspeccionFormPayload, IInspeccionListResponse } from '@/types/inspeccion'
import type { IParticipante, IParticipanteListResponse } from '@/types/participante'

export interface InspeccionListParams {
  page?: number
  per_page?: number
  search?: string
}

export async function getInspecciones(
  params: InspeccionListParams = {},
): Promise<IInspeccionListResponse> {
  const { data } = await api.get<IInspeccionListResponse>('/inspecciones', { params })
  return data
}

export async function createInspeccion(payload: IInspeccionFormPayload): Promise<IInspeccion> {
  const { data } = await api.post<{ data: IInspeccion }>('/inspecciones', payload)
  return data.data
}

export async function getVencimientosCarne(
  params: InspeccionListParams = {},
): Promise<IParticipanteListResponse> {
  const { data } = await api.get<IParticipanteListResponse>('/inspecciones/vencimientos-carne', {
    params,
  })
  return data
}

export async function getReinspecciones(
  params: InspeccionListParams = {},
): Promise<IInspeccionListResponse> {
  const { data } = await api.get<IInspeccionListResponse>('/inspecciones/reinspecciones', {
    params,
  })
  return data
}

export interface ParticipanteCarnePayload {
  numero_carne: string | null
  fecha_emision_carne: string | null
  fecha_vencimiento_carne: string | null
}

export async function updateParticipanteCarne(
  participanteId: number,
  payload: ParticipanteCarnePayload,
): Promise<IParticipante> {
  const { data } = await api.patch<{ data: IParticipante }>(
    `/participantes/${participanteId}/carne`,
    payload,
  )

  return data.data
}

File: ferias_app/lib/services/inspeccion_service.dart
import '../models/inspeccion.dart';
import '../models/paginated_response.dart';
import '../models/participante.dart';
import 'api_service.dart';

class InspeccionService {
  InspeccionService({ApiService? apiService})
    : _apiService = apiService ?? ApiService();

  final ApiService _apiService;

  Future<PaginatedResponse<Inspeccion>> getInspecciones({
    String? search,
    int page = 1,
    int perPage = 10,
  }) async {
    final response = await _apiService.get<Map<String, dynamic>>(
      '/inspecciones',
      queryParameters: <String, dynamic>{
        'search': search?.trim().isEmpty == true ? null : search?.trim(),
        'page': page,
        'per_page': perPage,
      }..removeWhere((key, value) => value == null),
    );

    return PaginatedResponse<Inspeccion>.fromJson(
      response.data ?? <String, dynamic>{},
      Inspeccion.fromJson,
    );
  }

  Future<PaginatedResponse<Inspeccion>> getReinspecciones({
    String? search,
    int page = 1,
    int perPage = 10,
  }) async {
    final response = await _apiService.get<Map<String, dynamic>>(
      '/inspecciones/reinspecciones',
      queryParameters: <String, dynamic>{
        'search': search?.trim().isEmpty == true ? null : search?.trim(),
        'page': page,
        'per_page': perPage,
      }..removeWhere((key, value) => value == null),
    );

    return PaginatedResponse<Inspeccion>.fromJson(
      response.data ?? <String, dynamic>{},
      Inspeccion.fromJson,
    );
  }

  Future<PaginatedResponse<Participante>> getVencimientosCarne({
    String? search,
    int page = 1,
    int perPage = 15,
  }) async {
    final response = await _apiService.get<Map<String, dynamic>>(
      '/inspecciones/vencimientos-carne',
      queryParameters: <String, dynamic>{
        'search': search?.trim().isEmpty == true ? null : search?.trim(),
        'page': page,
        'per_page': perPage,
      }..removeWhere((key, value) => value == null),
    );

    return PaginatedResponse<Participante>.fromJson(
      response.data ?? <String, dynamic>{},
      Participante.fromJson,
    );
  }

  Future<Inspeccion> createInspeccion({
    required Map<String, dynamic> data,
  }) async {
    final response = await _apiService.post<Map<String, dynamic>>(
      '/inspecciones',
      data: data,
    );

    return _fromResource(response.data);
  }

  Inspeccion _fromResource(Map<String, dynamic>? payload) {
    final data = payload?['data'];

    if (data is Map) {
      return Inspeccion.fromJson(Map<String, dynamic>.from(data));
    }

    return Inspeccion.fromJson(payload ?? <String, dynamic>{});
  }
}
